### PR TITLE
defined the types used from the spotify object model

### PR DESCRIPTION
**Pull request by janeson332**
are for this structs ok, or should i use classes instead?

### CODE REVIEW

**Comment by skaupper** [comment]
i dont mind structs at all. maybe without the typedef prefix haha

**Comment by janeson332** [comment]
better ? :D

**Comment by skaupper** [comment]
jep :)

File: src/SpotifyApi/SpotifyAPI.cpp [hunk 1]
```diff
@@ -13,6 +18,29 @@ TResult<Token> SpotifyAPI::getAccessToken(GrantType grantType,
                                           std::string const &redirectUri,
                                           std::string clientID,
                                           std::string clientSecret) {
+
+  // only authorization code supported until now ..
+  assert(grantType == AuthorizationCode);
+  auto client = std::make_unique<RestClient::Connection>(cSpotifyBaseUrl);
+
+  // build body
+  std::string body;
```
Comment (skaupper) [inline src/SpotifyApi/SpotifyAPI.cpp]: take a look into `std::stringstream`

Comment (skaupper) [reply src/SpotifyApi/SpotifyAPI.cpp]: (nice to know)

Comment (janeson332) [reply src/SpotifyApi/SpotifyAPI.cpp]: haha, yeah i know ^^

**Pull request merged**